packages/medusa/src/strategies/cart-completion.ts

```diff
@@ -299,19 +301,12 @@ class CartCompletionStrategy extends AbstractCartCompletionStrategy {
 
     let order: Order
     try {
-      order = await this.orderService_
-        .withTransaction(manager)
-        .createFromCart(cart)
+      order = await orderServiceTx.createFromCart(cart)
     } catch (error) {
       if (error && error.message === ORDER_CART_ALREADY_EXISTS_ERROR) {
-        order = await this.orderService_
-          .withTransaction(manager)
-          .retrieveByWithTotals(
-            { cart_id: id },
-            {
-              relations: ["shipping_address", "payments"],
-            }
-          )
+        order = await orderServiceTx.retrieveByCartId(id, {
+          relations: ["shipping_address", "payments"],
+        })
```

Comment:
thought: do we need to decorate totals here too?

Reply:
No, I need to find a way to split that method to get the raw result but i am on another thing at the moment 👍